applications/CoSimulationApplication/tests/dummy_solver/cpp/dummy_solver.cpp

```diff
@@ -0,0 +1,327 @@
+        InitializeSolutionStep();
+        Predict();
+        SolveSolutionStep();
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+}
+
+void RunSolutionLoopWithWeakCoupling(MeshType& rMesh, DataFieldType& rDataField)
+{
+    // Note the following only works with one coupling inteface, requires more effort to make it work with multiple coupling interfaces.
+
+    CoSim::CoSimIO co_sim_io("dummy_solver_cpp", "dummy_solver_io_settings");
+
+    co_sim_io.Connect();
+    ExportMesh(co_sim_io, rMesh, "interface"); // send the coupling-interface mesh to be used for e.g. mapping
+
+    double current_time = 0.0;
+    const double end_time = 0.49;
+    while (current_time<end_time) {
+        current_time = AdvanceInTime(current_time);
+        InitializeSolutionStep();
+        Predict();
+
+        ImportData(co_sim_io, rDataField, "interface_temp");
+        SolveSolutionStep();
+        ExportData(co_sim_io, rDataField, "interface_pressure");
+
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+
+    co_sim_io.Disconnect();
+}
+
+void RunSolutionLoopWithStrongCoupling(MeshType& rMesh, DataFieldType& rDataField)
+{
+    // Note the following only works with one coupling inteface, requires more effort to make it work with multiple coupling interfaces.
+
+    CoSim::CoSimIO co_sim_io("dummy_solver_cpp", "dummy_solver_io_settings");
+
+    co_sim_io.Connect();
+    ExportMesh(co_sim_io, rMesh, "interface"); // send the coupling-interface mesh to be used for e.g. mapping
+
+    int control_signal;
+    std::string identifier;
+
+    double current_time = 0.0;
+    const double end_time = 0.49;
+    while (current_time<end_time) {
+        current_time = AdvanceInTime(current_time);
+        InitializeSolutionStep();
+        Predict();
+        while(true) {
+            ImportData(co_sim_io, rDataField, "interface_temp");
+            SolveSolutionStep();
+            ExportData(co_sim_io, rDataField, "interface_pressure");
+            control_signal = co_sim_io.RecvControlSignal(identifier);
+            if (control_signal == 51) { // convergence achieved
+                break;
+            }
+        }
+
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+
+    co_sim_io.Disconnect();
+}
+
+void RunSolutionCoSimulationOrchestrated(MeshType& rMesh, DataFieldType& rDataField)
+{
+    CoSim::CoSimIO co_sim_io("dummy_solver_cpp", "dummy_solver_io_settings");
+
+    co_sim_io.Connect();
+
+    int control_signal;
+    std::string identifier;
+    while(true) {
+        // receive control signal to decide what to do
+        // the signals are defined in KratosMultiphysics/applications/CoSimulationApplication/python_scripts/co_simulation_tools.py
+        control_signal = co_sim_io.RecvControlSignal(identifier);
+        if (control_signal == 1) {
```

Comment:
I assume that these control signals should be given with some constexp or enums.

Reply:
solved